### ISSUE TITLE
Make Fork use the custom diff driver command defined in .gitconfig

### ISSUE THREAD

**Issue by AnomalousUnderdog** [ForkIssues/TrackerWin#1566]
I have some binary files that represent tiles in a 2D grid. I made a [custom console application](https://github.com/AnomalousUnderdog/AreaMapTilesTextConv) that displays it into a textual format to standard output, usable for textconv. Fork already respects the [textconv](https://git-scm.com/docs/diff-config#Documentation/diff-config.txt-diffltdrivergttextconv) option, but it was rather limiting.

The only argument passed when using textconv is the file path (which could be in a temp folder).

However my custom console application needed the original path of the file (as it was in the repo), because I want it to also display data found in an accompanying json text file (i.e. labels attached to certain tiles). That json text file is found in the same path as the binary file (e.g. a `Highway.AreaMapTiles.bytes` file will have a `Highway.AreaMap.txt` in the same path). The problem is, with the textconv way, the "old" file turns out to be in a temp folder, which means it has no way of getting the json text file.

Notice the first line in this side-by-side diff from Fork, the old-file (left side) is in a temp folder (this is from when I was using `textconv`):
![Fork_aYIUGOJ8sG](https://user-images.githubusercontent.com/553006/174349313-ce0085e8-6668-4a00-ab66-8d6d28cc7a85.png)

From what I researched, the proper way to do what I want is using the [command](https://git-scm.com/docs/diff-config#Documentation/diff-config.txt-diffltdrivergtcommand) option.

So I changed my console application to accept the 7 arguments as detailed in [GIT_EXTERNAL_DIFF](https://git-scm.com/docs/git#Documentation/git.txt-codeGITEXTERNALDIFFcode) of the docs. And what I did was, instead of having to do the code for outputting diff text, I convert both the old-file and new-file into text format, save them to temporary text files, and do a `git diff` on them. I then redirect the output of git diff into my application's standard output.

Now my `.gitconfig` uses `command` instead of `textconv`.

It works properly when I use git from bash, for example in my repo I do `git diff "Chosen One's Home.AreaMapTiles.bytes"`:
![bash_92sao7QQRI](https://user-images.githubusercontent.com/553006/174347436-881e1cfb-71eb-4d5e-9e0c-25fe26e84075.png)

But when I view the same file in Fork, it seems to completely ignore the `command` option.
![Fork_4MUh1YfUrl](https://user-images.githubusercontent.com/553006/174347462-abc64b7a-46e8-43cd-8434-2b9082fb64c0.png)

The only git GUI client I've seen that works properly is Git Extensions (I do prefer Fork since Git Extensions can't show side-by-side diff):
![GitExtensions_S5ydK9SwCt](https://user-images.githubusercontent.com/553006/174347472-cafa7e1e-f6a6-4943-bd54-09e6c78e83e3.png)

This is in my `.gitconfig`:
```
[diff "AreaMapTilesBytes"]
	command = B:/Projects/AreaMapTilesTextConv/AreaMapTilesTextConv/bin/Release/AreaMapTilesTextConv.exe
```

Disregard the rather long path to the program, it shouldn't really matter. That's just where my IDE outputs the executable as I'm working on it. I'll move it to a system path once I'm happy with it.

And this is in my `.gitattributes`:
```
*.AreaMapTiles.bytes diff=AreaMapTilesBytes
```

This is what that map looks like:
![Unity_nAff3e5flY](https://user-images.githubusercontent.com/553006/174348066-b5757154-3703-4b75-8064-7d638b6a37d3.jpg)

This is what that map looks like in the console application (without doing any diff):
```
=-=-=-=-============================================-=-=-=-=
Version = 2
Width = 52
Height = 47
OffsetLeft = -1
OffsetBottom = 0
=-=-=-=-============================================-=-=-=-=
. = COFFEEWALK8899.AreaMapTile = Default Walkable Tile
X = DEADBEEFIN6969.AreaMapTile = Default Wall Tile
# = LEFOYEVOHU9627.AreaMapTile = Low Cover
% = XIHEYUBIJU4285.AreaMapTile = LargeRock
* = DOXAMETUZI6843.AreaMapTile = Concrete Wall
- = ZUTADOMIBE5773.AreaMapTile = High Cover
=-=-=-=-============================================-=-=-=-=
  0123456789012345678901234567890123456789012345678901
6:....................................................
5:....................................................
4:..........................................#XX##X#...
3:..........................................XXXXXX#...
2:..............................##..........XXXXXXX#..
1:..............................XX..........XXXXXXX#..
0:..............................XX..........XXX####...
9:..............................XX..........###.......
8:..............................##....................
7:....................................................
6:....................................................
5:.......................XXXXXXXX.....................
4:...XXXXXXXXXXXXXXXXXXXXX......XXXXXXXXXXXXXXXXXXXXXX
3:...XX##........#....#...........####...............X
2:...XX##.........#...............####...............X
1:...XX##.........#...............####....XXXXXX#....X
0:...XX##.........####....................X..........X
9:...XX##.........####...XX..............##..........X
8:...XX##...........................X...........#....X
7:...XX##....................##XX##.#X#.#XX#....X....X
6:...XX##........#....#......#XXXX#.#X#.#XX#..##X#...X
5:...XX##............*******.#XXX##.#.X.......##X#...X
4:...XX#########.....*******..###....#.#...##..#X##..X
3:...XXXXXXXXXXXXXXX.*******........X##.X.#XXXXXX##..X
2:...X...............*******.........................X
1:...X...........--..................................X
0:...X...........--...XXXXXXXXXXXXXXXX...............X
9:...X..#..#..#..--.##X..............X...#...#....#..X
8:...X...........--.#....................#...#....#..X
7:...X...........--.#.X..............X...#...#....#..X
6:...X..#..#..#..--.#.XXXXXX.XXX.XXXXX...#...#....#..X
5:...X...........--.#....##.........X....#...#....#..X
4:...X.................##XXX.....XXX#....#...#....#..X
3:...X##########........#XX...X..XXX.....#...#....#..X
2:...X.##......%%%%%%.#.#XX...X.#XX#.....#...#....#..X
1:...X.XX#.....%%%%%%...#X....X.#XX#.....#...#....#..X
0:...X.X#.....#%%%%%%....X....X..XX#.....#...#....#..X
9:...X.X#......%%%%%%....X..X#X..XX#.....#...#....#..X
8:...X.X..#..............#..XXXXXX.......#...#....#..X
7:...X.X..X.###..........#.......#...................X
6:...X.X#.X.#.....#......#...........................X
5:...X.XXXX.##....#......#...........................X
4:...X......##....#......#########..............###..X
3:...X......#.....#.............................###..X
2:...X......#######.......................###........X
1:...X............................X#XXX#..###........X
0:...XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
=-=-=-=-============================================-=-=-=-=
```

Once again, here is a link to my [console application](https://github.com/AnomalousUnderdog/AreaMapTilesTextConv). It also has test files in them like the one in my screenshots in case anyone wants to look at it.

**Comment by DanPristupov** [comment]
What git instance do you use in Fork? Did you try to switch it to your system git?

**Comment by DanPristupov** [comment]
One more question: could you check whether 9968 bytes is it the binary or text file of your file?

**Comment by AnomalousUnderdog** [comment]
@DanPristupov Thanks for the response. git instance was already my system one, 2.36.1.windows.1 I've always been using that. Switching between Fork's git or my system git doesn't seem to change the outcome.

The 9968 bytes is definitely the binary file. The temporary text files don't get created at all when looking at the binary files through Fork.

**Comment by craigbrown** [comment]
Did you manage to get this working?

I'm having a similar issue. I'm trying to use the [git-xl](https://github.com/xlwings/git-xl) extension, which let you diff an Excel file to see VBA changes. This appears to hook in to git diff in the same way. It works fine on the command line in the shell that Fork is using, but not within Fork.

My guess is that this is because of the file paths displayed at the top of the diff. Fork probably expects those (or at least one?) to match the path of the file you've selected. Whereas git-xl makes up some paths to represent the VBA modules inside an Excel file, and it looks like your tool is using a temp path.

**Comment by DanPristupov** [comment]
I don't know how those extensions work, but you can try to use them using `.gitattributes` file.

There is an example from the official `git-for-windows` distribution, which shows diff for `.doc` and `.pdf` files:
```
###############################################################################
# diff behavior for common document formats
# 
# Convert binary document formats to text before diffing them. This feature
# is only available from the command line. Turn it on by uncommenting the 
# entries below.
###############################################################################
*.doc   diff=astextplain
*.docx  diff=astextplain
```

`astextplain` is a bash script in `%localappdata%\Fork\gitInstance\2.39.1\usr\bin\astextplain` (open it in some editor and check how it's implemented).

**Comment by craigbrown** [comment]
Yeah I have the `.gitattributes` file all set up correctly I think. It's working on the command line.

I modified the extension to log to a file when it gets called by git diff. The log gets written when called from the command line (using the same shell/git as Fork). Nothing when I try to view the diff in Fork.

Maybe Fork is doing something clever with git diff which means these diff drivers get bypassed, or not calling it all for a big binary file.

**Comment by CarlAndersson** [comment]
I have very similar problems. Everything working on command line, but fork does not seem to use these.
